railties/lib/rails/application/routes_reloader.rb

```diff
@@ -44,7 +49,13 @@ def clear!
       end
 
       def load_paths
-        paths.each { |path| load(path) }
+        run_callbacks :load_paths do
+          paths.each { |path| load(path) }
+        end
```

Comment:
Do we need the callback setup feels like we could just do this?

```ruby
def load_paths
  paths.each { |path| load(path) }
  after_load_path_callback&.call
end
```

Reply:
@rafaelfranca had suggested this but it might very well be faster not to use the activesupport callbacks? I have no strong feelings either way and am happy to make changes

Reply:
I'll remove it after the merge